src/collective/solr/solr.py

```diff
@@ -225,11 +251,24 @@ def add(self, boost_values=None, **fields):
         else:
             lst.append('<doc>')
         for f, v in fields.items():
+
+            # Add update="set" attribute to each field except for the uniqueKey
+            # field.
+            if f == uniqueKey:
+                tmpl = '<field name="%s">%%s</field>' % self.escapeKey(f)
+                lst.append(tmpl % self.escapeVal(v))
+                continue
+
             if f in boost_values:
-                tmpl = '<field name="%s" boost="%s">%%s</field>' % (
+                tmpl = '<field name="%s" boost="%s" update="set">%%s</field>' % (
```

Comment:
The `update="set"` Attribut will be removed later, if atomic updates are enabled. And yes you're if this attribute is present the index-time-boosting feature is disabled.  --> https://github.com/collective/collective.solr/pull/86/files#diff-ce14634d17ed2770c78538cdc89f0fcdR268